trunner/target/armv7m4.py

```diff
@@ -27,28 +28,47 @@
 
 class ARMv7M4Rebooter(Rebooter):
     # TODO add text mode
+    # NOTE: changing boot modes not needed/supported for this target
     pass
 
 
-class OpenocdGdbServerHarness(HarnessBase):
-    def __call__(self):
+class STM32L4x6OpenocdGdbServerHarness(IntermediateHarness):
+    """Harness that runs other harness in Opendocd gdb server context.
+
+    It calls passed harness under gdb server context and then continue normal
+    harness execution flow.
+
+    Attributes:
+        harness: Harness that will be run under gdb server context.
+
+    """
+
+    def __init__(self, harness: HarnessBase):
+        super().__init__()
+        self.harness = harness
+
+    def __call__(self) -> Optional[TestResult]:
         with OpenocdGdbServer(interface="stlink", target="stm32l4x").run():
-            return self.harness()
+            self.harness()
+
+        return self.next_harness()
 
 
-class STM32L4x6PloAppLoader(HarnessBase, PloInterface):
+class STM32L4x6PloAppLoader(TermHarness, PloInterface):
     def __init__(self, dut: Dut, apps: Sequence[AppOptions], gdb: GdbInteractive):
+        TermHarness.__init__(self)
+        PloInterface.__init__(self, dut)
         self.dut = dut
         self.apps = apps
         self.gdb = gdb
         self.load_offset = 0x30000
         self.ram_addr = 0x20000000
         self.page_sz = 0x200
-        super().__init__()
 
     def _app_size(self, path: Path):
         sz = path.stat().st_size
-        offset = self.page_sz - (sz % self.page_sz) if sz % self.page_sz != 0 else 0
+        # round up to the size of the page
+        offset = (self.page_sz - sz) % self.page_sz
```

Comment:
(-sz) % self.page_sz would be enough actually, however I am not sure what's more readable

Reply:
I think the first one is more readable